Doing most of my todos in splitquery

erzel opened the pull request:
1. Added checks that for FULL_SCAN algorithm the split_columns are the primary key and must be the prefix of an index in general.
2. Added creation of SplitParams with either SplitCount or NumRowsPerQueryPart and allowed computing one from the other.
3. SplitColumns now default to primary key
4. Replaced assertion-functions with "if (!cond) { panic ... }"
@michael-berlin @dumbunny please take a look.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/youtube/vitess/1592)
<!-- Reviewable:end -->

alainjobart commented on go/vt/tabletserver/splitquery/equal_splits_algorithm.go:
```diff
@@ -38,7 +38,10 @@ type EqualSplitsAlgorithm struct {
 
-	assertEqual(len(splitParams.splitColumns), len(splitParams.splitColumnTypes))
+	if len(splitParams.splitColumns) != len(splitParams.splitColumnTypes) {
+		panic(fmt.Sprintf("len(splitparams.splitColumns) != len(splitparams.splitColumnTypes): %v!=%v",
```
I thought we agreed bad user input should return an error, not panic, if possible?

michael-berlin commented:
Reviewed 4 of 4 files at r1, 10 of 10 files at r2.
Review status: all files reviewed at latest revision, 27 unresolved discussions.

---

*[go/vt/tabletserver/splitquery/doc.go, line 12 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdUyvB5OBpwVwq9IzO:-KDdUyvB5OBpwVwq9IzP:-386750623) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/doc.go#L12)):*
"id" ?

It looks to me that this code would not compile in a real world example.

You should consider to add a Go example which will always be run by the unit tests: https://blog.golang.org/examples

The example will also show up in the documentation then.

The only downside is that you probably have to mock out everything e.g. MySQLExecutor

---

*[go/vt/tabletserver/splitquery/doc.go, line 17 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdXbynEU-M1LlJBjlH:-KDdXbynEU-M1LlJBjlI:-1008918853) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/doc.go#L17)):*
It would be nice to have a very short reference here how to get this schema?

---

*[go/vt/tabletserver/splitquery/doc.go, line 18 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdV5xIZYo1DFrwEERL:-KDdV5xIZYo1DFrwEERM:-729371647) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/doc.go#L18)):*
Remove ( ) because they are not necessary in Go.

Same for the examples below.

---

*[go/vt/tabletserver/splitquery/doc.go, line 19 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdVQGhKQ045AcXeAVK:-KDdVQGhKQ045AcXeAVL:1667460104) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/doc.go#L19)):*
return fmt.Errorf("SplitParams creation failed: %v", err)

On this note: In general, error messages should always start with a lower case string because they may be chained: "layer1-error: layer2-error: layer3-error"

But in case of capitalized names this is fine. So "SplitParams" instead of "splitParams" is correct here.

---

*[go/vt/tabletserver/splitquery/doc.go, line 32 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdX6olvz-max-84maQ:-KDdX6oms98w6hidG9bd:-1536066458) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/doc.go#L32)):*
Go requires you to keep the case of abbreviations e.g. write "getID" instead of "getId".

In this case, "Sql" should be all uppercase. But I admit that "mySQLExecutor" looks ugly too :)

---

*[go/vt/tabletserver/splitquery/doc.go, line 34 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdVdqdrUNMiANVU6Qy:-KDdVdqdrUNMiANVU6Qz:-958373658) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/doc.go#L34)):*
return fmt.Errorf("FullScanAlgorithm creation failed: %v", err)

---

*[go/vt/tabletserver/splitquery/doc.go, line 37 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdVriuyjTqEkWxUqgf:-KDdVriuyjTqEkWxUqgg:-1206708271) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/doc.go#L37)):*
Remove "Always succeeds" because it's not relevant. If it would return an error, the example below would throw a compile error.

---

*[go/vt/tabletserver/splitquery/doc.go, line 45 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdW2neYHo3StgoMGFm:-KDdW2neYHo3StgoMGFn:1610952934) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/doc.go#L45)):*
No need to declare queryParts.

Just write:

queryParts, err := splitter.Split()

---

*[go/vt/tabletserver/splitquery/doc.go, line 47 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdVkBhzPCx2TPiNvwq:-KDdVkBi3WlYT_4HRx3x:503490270) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/doc.go#L47)):*
return fmt.Errorf("splitter.Split() failed: %v", err)

---

*[go/vt/tabletserver/splitquery/split_params.go, line 39 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdYTkrfUGkL1yzpgBQ:-KDdYTks35JLp1gbEtTb:-1154574410) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/split_params.go#L39)):*
nit: s/vitess/Vitess/

---

*[go/vt/tabletserver/splitquery/split_params.go, line 40 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdYd9pPt3grXoiN85W:-KDdYd9pPt3grXoiN85X:-2078659799) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/split_params.go#L40)):*
Vitess

---

*[go/vt/tabletserver/splitquery/split_params.go, line 50 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdYncVDsxP71H4oYzL:-KDdYncVDsxP71H4oYzM:1511881628) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/split_params.go#L50)):*
"is the list"

---

*[go/vt/tabletserver/splitquery/split_params.go, line 54 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdYvTHAN9wwy_SdQlK:-KDdYvTHAN9wwy_SdQlL:-1176756880) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/split_params.go#L54)):*
"is the desired"

---

*[go/vt/tabletserver/splitquery/split_params.go, line 66 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdZ7Rz0poeFvOo0DC_:-KDdZ7Rz0poeFvOo0DCa:-27613499) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/split_params.go#L66)):*
non-negative?

---

*[go/vt/tabletserver/splitquery/split_params.go, line 74 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdZO5Cl1FubkdmeuW2:-KDdZO5Cl1FubkdmeuW3:1291066823) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/split_params.go#L74)):*
Just as a note: There's a reason that Go doesn't provide such a simple function in their libraries :(

They argue that you should just write this out instead of using a function.

However, I'm fine with using the function here, but just that you know.

---

*[go/vt/tabletserver/splitquery/split_params.go, line 79 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdZseMTcFBb3uaq-m8:-KDdZseMTcFBb3uaq-m9:-2078659799) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/split_params.go#L79)):*
Vitess

---

*[go/vt/tabletserver/splitquery/split_params.go, line 80 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdZuHSWlObG9-dDil8:-KDdZuHSWlObG9-dDil9:-2078659799) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/split_params.go#L80)):*
Vitess

---

*[go/vt/tabletserver/splitquery/split_params.go, line 90 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdZx5OQHbRRwFDyoB_:-KDdZx5OQHbRRwFDyoBa:1312917246) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/split_params.go#L90)):*
is the list

---

*[go/vt/tabletserver/splitquery/split_params.go, line 94 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdZypx24V-Redm1PJM:-KDdZypx24V-Redm1PJN:-1527959104) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/split_params.go#L94)):*
is the

---

*[go/vt/tabletserver/splitquery/split_params.go, line 107 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDd_0hNdNrusymv0EuJ:-KDd_0hNdNrusymv0EuK:-27613499) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/split_params.go#L107)):*
non-negative?

---

*[go/vt/tabletserver/splitquery/splitter.go, line 150 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDd_tJSywDs3K7gCofg:-KDd_tJSywDs3K7gCofh:820413743) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/splitter.go#L150)):*
represents?

---

*[go/vt/tabletserver/splitquery/splitter.go, line 151 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDd_xVNdQtvMHAatmNG:-KDd_xVNdQtvMHAatmNH:-468164802) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/splitter.go#L151)):*
false

---

*[go/vt/tabletserver/splitquery/splitter.go, line 157 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdaVmuYNmtPsgYupqN:-KDdaVmuYNmtPsgYupqO:1309325789) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/splitter.go#L157)):*
This paragraph should be on top of the comment after the initial one-line comment.

It helped me a lot to understand the purpose of this function.

---

*[go/vt/tabletserver/splitquery/splitter.go, line 166 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdc7y8RlK4kgRcsxG0:-KDdc7y8RlK4kgRcsxG1:1434114251) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/splitter.go#L166)):*
This is also no user input?

---

*[go/vt/tabletserver/splitquery/splitter.go, line 173 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdbMRSvjXYQ8GkRm1w:-KDdbMRSvjXYQ8GkRm1x:-1129185616) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/splitter.go#L173)):*
You have an extra Unchecked method just for saving the assert checks when you do further recursive calls?

That's probably not necessary?

---

*[go/vt/tabletserver/splitquery/splitter.go, line 188 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdbscsYttvrhmMm0Lj:-KDdbscsYttvrhmMm0Lk:-460664648) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/splitter.go#L188)):*
needs

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/youtube/vitess/1592)*
<!-- Sent from Reviewable.io -->

michael-berlin commented:
LGTM after comments are addressed.

Let me know if you have questions.

erzel commented:
Review status: all files reviewed at latest revision, 27 unresolved discussions.

---

*[go/vt/tabletserver/splitquery/doc.go, line 12 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdUyvB5OBpwVwq9IzO:-KDdutNLwy9scYLw8iLz:-499146009) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/doc.go#L12)):*
Done.

---

*[go/vt/tabletserver/splitquery/doc.go, line 17 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdXbynEU-M1LlJBjlH:-KDdvh-FG0cq3S-IoCu2:-499146009) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/doc.go#L17)):*
Done.

---

*[go/vt/tabletserver/splitquery/doc.go, line 18 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdV5xIZYo1DFrwEERL:-KDdvih7onaWxfeaPt4I:-499146009) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/doc.go#L18)):*
Done.

---

*[go/vt/tabletserver/splitquery/doc.go, line 32 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdX6olvz-max-84maQ:-KDdvwiwAAPrkCrrTxXT:-13693215) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/doc.go#L32)):*
Changed this to be a function adhering to the style.

---

*[go/vt/tabletserver/splitquery/doc.go, line 34 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdVdqdrUNMiANVU6Qy:-KDdw1RWT5f9Wz7qDu9X:-499146009) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/doc.go#L34)):*
Done.

---

*[go/vt/tabletserver/splitquery/doc.go, line 37 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdVriuyjTqEkWxUqgf:-KDdw3yXArXtpKCwyJxs:-499146009) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/doc.go#L37)):*
Done.

---

*[go/vt/tabletserver/splitquery/doc.go, line 45 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdW2neYHo3StgoMGFm:-KDdw9SCO19PdtD5YDTM:329533245) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/doc.go#L45)):*
Wanted to make the return type explicit for the benefit of the example readers here.

---

*[go/vt/tabletserver/splitquery/doc.go, line 47 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdVkBhzPCx2TPiNvwq:-KDdwH7jtf0Hdcqj6wC8:-499146009) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/doc.go#L47)):*
Done.

---

*[go/vt/tabletserver/splitquery/equal_splits_algorithm.go, line 42 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KD_wyZonC_uSk8eiZdg-r2-42:-KDdwHwHApL6bVg7QH87:-1390707149) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/equal_splits_algorithm.go#L42)):*
Note that this is not bad user input. The SplitParams struct maintains an array of SplitColumnTypes whose elements correspond to the SplitColumns (the user does not specify the types-they are computed from the SplitColumns using the schema). One of the invariants of SplitParams struct is that  the length of these two slices is always the same. Hence this check.

I'm not sure to which comment below you are referring.

---

*[go/vt/tabletserver/splitquery/split_params.go, line 39 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdYTkrfUGkL1yzpgBQ:-KDdxENAOBCEDd0IC1LQ:-499146009) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/split_params.go#L39)):*
Done.

---

*[go/vt/tabletserver/splitquery/split_params.go, line 50 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdYncVDsxP71H4oYzL:-KDdxJELzDAZwARDOJUy:-499146009) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/split_params.go#L50)):*
Done.

---

*[go/vt/tabletserver/splitquery/split_params.go, line 54 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdYvTHAN9wwy_SdQlK:-KDdxMYRVafFAh9rZlja:-499146009) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/split_params.go#L54)):*
Done.

---

*[go/vt/tabletserver/splitquery/split_params.go, line 66 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdZ7Rz0poeFvOo0DC_:-KDdxRjpWvieCf-pemBg:-499146009) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/split_params.go#L66)):*
Done.

---

*[go/vt/tabletserver/splitquery/split_params.go, line 74 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdZO5Cl1FubkdmeuW2:-KDdxVpbTXo8LtAaczH1:-984335159) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/split_params.go#L74)):*
Yeah I looked into that.... I found a thread where they argue that you should write such a function per project.
So I did. :-).

---

*[go/vt/tabletserver/splitquery/split_params.go, line 79 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdZseMTcFBb3uaq-m8:-KDdxhbQm6-thqwwXlPl:-499146009) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/split_params.go#L79)):*
Done.

---

*[go/vt/tabletserver/splitquery/split_params.go, line 90 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdZx5OQHbRRwFDyoB_:-KDdyEkjQzlpEVsmFPkw:-499146009) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/split_params.go#L90)):*
Done.

---

*[go/vt/tabletserver/splitquery/split_params.go, line 94 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdZypx24V-Redm1PJM:-KDdyETe96ZJ-H-Ir1Yf:-499146009) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/split_params.go#L94)):*
Done.

---

*[go/vt/tabletserver/splitquery/split_params.go, line 107 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDd_0hNdNrusymv0EuJ:-KDdyP2cFvJYHPrgC3fz:-964681822) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/split_params.go#L107)):*
It should actually be positive. I don't know what I was thinking.

---

*[go/vt/tabletserver/splitquery/splitter.go, line 150 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDd_tJSywDs3K7gCofg:-KDdyh5XAcybQ8gq4Gk9:-499146009) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/splitter.go#L150)):*
Done.

---

*[go/vt/tabletserver/splitquery/splitter.go, line 151 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDd_xVNdQtvMHAatmNG:-KDdykpYV1Tg9hiaOpAY:-499146009) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/splitter.go#L151)):*
Done.

---

*[go/vt/tabletserver/splitquery/splitter.go, line 157 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdaVmuYNmtPsgYupqN:-KDdz9z07ZBxb14_owkO:-499146009) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/splitter.go#L157)):*
Done.

---

*[go/vt/tabletserver/splitquery/splitter.go, line 166 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdc7y8RlK4kgRcsxG0:-KDdzY1MDB1q8Uh5grja:-899052233) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/splitter.go#L166)):*
Yes. This non-exported method should always be called correctly; otherwise it's a programming error.

---

*[go/vt/tabletserver/splitquery/splitter.go, line 173 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdbMRSvjXYQ8GkRm1w:-KDdzrEETcuQqXWqDx14:1450394838) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/splitter.go#L173)):*
Yes. I converted the second function to a "nested" function of the first one to make it more obvious (and safe).

---

*[go/vt/tabletserver/splitquery/splitter.go, line 188 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdbscsYttvrhmMm0Lj:-KDe1nISLe_w541UPhGe:-499146009) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/splitter.go#L188)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/youtube/vitess/1592)*
<!-- Sent from Reviewable.io -->

erzel commented:
Review status: 8 of 12 files reviewed at latest revision, 27 unresolved discussions.

---

*[go/vt/tabletserver/splitquery/equal_splits_algorithm.go, line 42 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KD_wyZonC_uSk8eiZdg-r2-42:-KDe3_LsHuAEoB1wQpPf:1885953511) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/equal_splits_algorithm.go#L42)):*
@alainjobart Please let me know if you have other comments or if I can submit. Thanks!

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/youtube/vitess/1592)*
<!-- Sent from Reviewable.io -->

michael-berlin commented:
Review status: 8 of 12 files reviewed at latest revision, 27 unresolved discussions.

---

*[go/vt/tabletserver/splitquery/doc.go, line 45 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdW2neYHo3StgoMGFm:-KDeezRt72pp9yduue__:-1439937249) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/doc.go#L45)):*
I thought so. But you already did that above when you wrote "slice of querytypes.QuerySplit objects".

Therefore, I would prefer to stick to idiomatic Go.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/youtube/vitess/1592)*
<!-- Sent from Reviewable.io -->

alainjobart commented:
LGTM

erzel commented:
Review status: 8 of 12 files reviewed at latest revision, 27 unresolved discussions.

---

*[go/vt/tabletserver/splitquery/doc.go, line 45 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdW2neYHo3StgoMGFm:-KDjAqZ62WqZqhyilHnm:-722506349) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/doc.go#L45)):*
Crap. I fixed it in my client, but forgot to push before I merged. Sorry. It'll be submitted in my next PR.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/youtube/vitess/1592)*
<!-- Sent from Reviewable.io -->

erzel commented:
Review status: 8 of 12 files reviewed at latest revision, 27 unresolved discussions.

---

*[go/vt/tabletserver/splitquery/split_params.go, line 80 \[r2\]](https://reviewable.io:443/reviews/youtube/vitess/1592#-KDdZuHSWlObG9-dDil8:-KDjB8488qiv6kcVJUAn:-499146009) ([raw file](https://github.com/youtube/vitess/blob/27e1e7e77679c1ab59dfd494b0db59a942b7c7ad/go/vt/tabletserver/splitquery/split_params.go#L80)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/youtube/vitess/1592)*
<!-- Sent from Reviewable.io -->
